### PR TITLE
[Finder] Update RealIteratorTestCase

### PR DESCRIPTION
**Pull request by flip111**
| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #27480
| License       | MIT
| Doc PR        | n/a

Makes the entire test directory empty instead of trying to delete particular files and directories. The old method failed when trying to remove a directory which was not empty.

### CODE REVIEW

File: src/Symfony/Component/Finder/Tests/Iterator/RealIteratorTestCase.php [hunk 1]
```diff
@@ -69,11 +69,23 @@ public static function setUpBeforeClass()
 
     public static function tearDownAfterClass()
     {
-        foreach (array_reverse(self::$files) as $file) {
-            if (DIRECTORY_SEPARATOR === $file[strlen($file) - 1]) {
-                @rmdir($file);
+        $paths = new \RecursiveIteratorIterator(
+                   new \RecursiveDirectoryIterator(
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Finder/Tests/Iterator/RealIteratorTestCase.php]: please use 4 spaces indentation

File: src/Symfony/Component/Finder/Tests/Iterator/RealIteratorTestCase.php [hunk 2]
```diff
@@ -69,11 +69,23 @@ public static function setUpBeforeClass()
...
+        foreach ($paths as $path) {
+            if ($path->isDir()) {
+                if ($path->isLink()) {
+                    unlink($path);
+                } else {
+                    rmdir($path);
+                }
             } else {
-                @unlink($file);
+                unlink($path);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Finder/Tests/Iterator/RealIteratorTestCase.php]: I'd suggest to keep the silencing operators (and add it above)

**Comment by flip111** [comment]
My current patch is based on master, should i open a new PR on 2.8 or what should i do about this?

**Comment by nicolas-grekas** [comment]
Thank you @flip111.

**Pull request merged**